From 426731f11fd5a423147be3a001ce713a19752ada Mon Sep 17 00:00:00 2001
From: Gerry Kopec <Gerry.Kopec@windriver.com>
Date: Wed, 15 Feb 2017 16:49:08 -0500
Subject: [PATCH] Add wrs-if properties to servers schema v2.37

In upstream Newton, api servers schema networks properties were
refactored to support automatic network allocation feature (aka "get
me a network").  This means that the wrs-if properties vif_model and
vif_pci_address have to be added back into the later version of the
create schema.

For next rebase, this commit should be merged with commits:
ee75aeb - Port critical Nova networking patches to Newton (vif_model)
c84512d - Port support for configurable pci bus slots for guest
          vnics (vif_pci_address)
---
 nova/api/openstack/compute/schemas/servers.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nova/api/openstack/compute/schemas/servers.py b/nova/api/openstack/compute/schemas/servers.py
index 6c0f6dd..ea5b8d8 100644
--- a/nova/api/openstack/compute/schemas/servers.py
+++ b/nova/api/openstack/compute/schemas/servers.py
@@ -110,6 +110,10 @@ base_create_v237['properties']['server']['properties']['networks'] = {
                                {'type': 'null'}]
                  },
                  'uuid': {'type': 'string', 'format': 'uuid'},
+                 # add vif_model and vif_pci_address to networks
+                 # properties for v2.37
+                 'wrs-if:vif_model': {'enum': model.VIF_MODEL_ALL},
+                 'wrs-if:vif_pci_address': {'type': 'string'},
              },
              'additionalProperties': False,
          },
-- 
2.7.4

